metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/dao/WorkflowUtils.java

```diff
@@ -323,42 +326,48 @@ public static AbstractExecutablePlugin computePredecessorPlugin(ExecutablePlugin
       ExecutablePluginType pluginType, ExecutablePluginType enforcedPredecessorType,
       String datasetId) throws PluginExecutionNotAllowed {
 
-    // If the plugin type does not need a predecessor (even the enforced one) we are done.
+    final PluginWithExecutionId<ExecutablePlugin> predecessorPlugin;
     final Set<ExecutablePluginType> defaultPredecessorTypes = getPredecessorTypes(pluginType);
-    if (defaultPredecessorTypes.isEmpty()) {
-      return null;
+    // If the plugin type does not need a predecessor (even the enforced one) we are done.
+    // We also return if a DEPUBLISH Operation is requested
+    if (defaultPredecessorTypes.isEmpty() || (defaultPredecessorTypes
+        .contains(ExecutablePluginType.PUBLISH) && pluginType == ExecutablePluginType.DEPUBLISH)) {
```

Comment:
Shouldn't we check that a valid publish exists? We could have someone triggering a depublish without there being a publish, right? We could execute the check in line 350-354 for that case as well? What do you think?